### PR TITLE
[receiver/splunkenterprise] 35445 add kvstore telemetry

### LINKED ISSUE #35445: [receiver/splunkenterprise] Adding Telemetry around KV Store
### Component(s)

receiver/splunkenterprise

### Is your feature request related to a problem? Please describe.

Currently, the Splunk Enterprise Receiver lacks the ability to report on the status of KV Store, KV Store replication, and KV Store backups. This information can be gathered from the `/services/kvstore/status` API endpoint in Splunk.

### Describe the solution you'd like

I would like to add metrics around the KV Store to splunkenterprisereceiver.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by shalper2**
#### Description
Add telemetry around KV Store status to the Splunk Enterprise Receiver.

#### Link to tracking issue
Fixes [35445](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35445)

#### Testing
Unit tests were updated and receiver component was deployed to test environments.

#### Documentation
Documentation was updated to reflect additional metrics

### CODE REVIEW

**Review by greatestusername — COMMENTED** [review]
This looks good to me!

Open question: Any other attributes that might be useful? instance id/guid or some other identifier?

**Comment by MovieStoreGuy** [comment]
> This looks good to me!
> 
> Open question: Any other attributes that might be useful? instance id/guid or some other identifier?

You could apply those as part of the resource or attribute processor, if you have any recommend values I would suggest adding it as an example configuration in the README :)

File: receiver/splunkenterprisereceiver/scraper.go [hunk 1]
```diff
@@ -1560,3 +1561,56 @@ func (s *splunkScraper) scrapeIntrospectionQueuesBytes(ctx context.Context, now
...
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		errs <- err
+		return
+	}
+
+	err = json.Unmarshal(body, &kvs)
```
Comment (MovieStoreGuy) [inline receiver/splunkenterprisereceiver/scraper.go:1598]: If you wanted, you can simplify this into:

```golang 
if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
   errs <- err
   return
}
```

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Are you able to add any tests to help validate that this works as expected?

File: receiver/splunkenterprisereceiver/scraper.go (hunk 1 above)
Comment (shalper2) [reply receiver/splunkenterprisereceiver/scraper.go]: I made this change and will probably repeat the pattern for the rest of the scrapers in a separate PR. Thanks!

**Comment by mx-psi** [comment]
> Any chance we can mark this as ready to merge?

There is a required test that is failing, we cannot merge until that is fixed

**Comment by shalper2** [comment]
@mx-psi should be all good to go now!

**Pull request merged**